homeassistant/components/nest/camera_sdm.py

```diff
@@ -193,10 +208,24 @@ async def _async_active_event_image(self):
         if not event_image:
             return None
         try:
-            image_bytes = await event_image.contents()
+            return await event_image.contents()
         except GoogleNestException as err:
             _LOGGER.debug("Unable to fetch event image: %s", err)
             return None
-        self._event_id = event_id
-        self._event_image_bytes = image_bytes
-        return image_bytes
+
+    def _schedule_event_image_cleanup(self, point_in_time):
+        """Schedules an alarm to remove the image bytes from memory, honoring expiration."""
+        if self._event_image_cleanup_unsub is not None:
+            self._event_image_cleanup_unsub()
+        self._event_image_cleanup_unsub = async_track_point_in_utc_time(
+            self.hass,
+            self._handle_event_image_cleanup,
+            point_in_time,
+        )
+
+    def _handle_event_image_cleanup(self, now):
+        """Clear images cached from events and scheduled callback."""
+        self._event_id = None
+        self._event_image_bytes = None
+        if self._event_image_cleanup_unsub is not None:
+            self._event_image_cleanup_unsub()
```

Comment:
This doesn't have to be called because track point in time is only ever called once. 

instead set it to `None`.
```suggestion
        self._event_image_cleanup_unsub = None
```